physical/raft/fsm.go

```diff
@@ -166,6 +164,9 @@ func (r *RaftBackend) SetFSMApplyCallback(f func()) {
 }
 
 func (f *FSM) openDBFile(dbPath string) error {
+	f.l.Lock()
```

Comment:
I think you need to modify Restore, since it grabs the lock and calls openDBFile.